presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/DistinctOutputQueryUtil.java

```diff
@@ -106,7 +106,7 @@ public Boolean visitFilter(FilterNode node, Void context)
         @Override
         public Boolean visitIntersect(IntersectNode node, Void context)
         {
-            return true;
+            return node.isDistinct() || node.getSources().stream().anyMatch(source -> lookupFunction.apply(source).accept(this, null));
```

Comment:
Trino uses allMatch here, which I believe is incorrect(takes care of less cases)